src/utils/getRowsInternal.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  GetRowsCallback,
+  GetRowsOptions,
+  GetRowsResponse,
+  TabularApiSurface,
+} from '../tabular-api-surface';
+import {createReadStreamInternal} from './createReadStreamInternal';
+import {Row} from '../row';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const concat = require('concat-stream');
+
+/**
+ * Get {@link Row} objects for the rows currently in your table.
+ *
+ * This method is not recommended for large datasets as it will buffer all rows
+ * before returning the results. Instead we recommend using the streaming API
+ * via {@link Table#createReadStream}.
+ *
+ * @param {TabularApiSurface} table The table instance to get rows from.
+ * @param {boolean} singleRow Boolean to check if the request is for a single row.
+ * @param {string} [viewName] The name of the authorized view, if applicable.
+ * @param {object} [optionsOrCallback] Configuration object. See
+ *     {@link Table#createReadStream} for a complete list of options.
+ * @param {object} [optionsOrCallback.gaxOptions] Request configuration options, outlined
+ *     here: https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {function} cb The callback function.
+ * @param {?error} cb.err An error returned while making this request.
+ * @param {Row[]} cb.rows List of Row objects.
+ *
+ * @returns {Promise<GetRowsResponse>|void} Returns a promise that resolves with the rows if no callback is provided, otherwise calls the callback with the rows.
+ *
+ * @example <caption>include:samples/api-reference-doc-snippets/table.js</caption>
+ * region_tag:bigtable_api_get_rows
+ */
+export function getRowsInternal(
```

Comment:
nit: It seems a but weird to have this helper that basically just defers to another helper, and is only used in a few places. I'd probably suggest to move this in with createReadStreamInternal (if not cut it out completely), since it seems like a pretty thin wrapper over it. I'm not sure if that would break a lot of tests or anything though

Reply:
These changes were introduced in https://github.com/googleapis/nodejs-bigtable/pull/1612.

This is necessary and here is why:
`table.getRows` calls `getRowsInternal`.
`table.createReadStream` calls `createReadStreamInternal`.

`getRowsInternal` and `createReadStreamInternal` both need a parameter telling them if the call is for a single row or for a set of rows. We cannot add this parameter to `table.getRows` or `table.createReadStream` because it changes the API surface.

Reply:
If `createReadStreamInternal` was pasted inline into `getRowsInternal` then `createReadStream` would have nothing to call.